pkg/cri/server/image_pull.go

```diff
@@ -369,12 +372,23 @@ func defaultScheme(host string) string {
 	if h, _, err := net.SplitHostPort(host); err == nil {
 		host = h
 	}
-	if host == "localhost" || host == "127.0.0.1" || host == "::1" {
+	if isLocalHost(host) {
 		return "http"
 	}
 	return "https"
 }
 
+// isLocalHost checks if the registry host is local.
+func isLocalHost(host string) bool {
+	if h, _, err := net.SplitHostPort(host); err == nil {
+		host = h
+	}
+	if host == "localhost" || host == "127.0.0.1" || host == "::1" {
```

Comment:
Can we use `net.IP.IsLoopback()` ? https://golang.org/pkg/net/#IP.IsLoopback

Reply:
There is `MatchLocalhost` which exists in `remotes/docker` package as well.  I'm not sure that parsing the string as an IP is worth it. Maybe for ipv6 it is tricky, for ipv4 we could instead just check for "127." prefix.

Reply:
> for ipv4 we could instead just check for "127." prefix.

Fun fact - `2130706433` is also a valid IPv4 loopback address 😛

Reply:
Updated to use `net.IP.IsLoopback()`